### DISCUSSION TITLE
MS Access Versions supported by UCanAccess

### DISCUSSION THREAD

**Discussion by bilaloumehdi** [spannm/ucanaccess#13] [Q&A] [answered]
Hello @spannm , I would like to ask about the support of MS access versions later than 2016. I saw that this not supported yet in 
https://ucanaccess.sourceforge.net/site.html. Could you please confirm that and is there will be a next version that supports newer versions ?

**Comment by spannm** [comment, marked as answer]
Microsoft Access databases, created using the Microsoft Access desktop application, with `*.accdb` extension are based on the so-called ACE format ("Access Connectivity Engine") and are forward- as well as backward-compatible between Access versions 2007, 2010, 2013, 2016, 2019, 2021 and so forth.

New Access versions may introduce new Access features. If a database makes use of such new feature, a field in the database file's binary header is set to indicate the minimum Access/ACE version required to fully support that particular database.

I am assuming that **all ACE-based database files are supported by recent versions of UCanAccess and Jackcess**.

Should you encounter an unsupported database file please file a bug report.

Further links:

- [Wikipedia: Microsoft Access JET / ACE Engine](https://en.wikipedia.org/wiki/Access_Database_Engine)
- [JetFormat class of Jackcess project](https://github.com/spannm/jackcess/blob/master/src/main/java/io/github/spannm/jackcess/impl/JetFormat.java)

Please note:
Your [Sourceforge link](https://ucanaccess.sourceforge.net/site.html) above is not edited by me. My project is a fork of UCanAccess as previously hosted at Sourceforge.
